opensrp-maternity/src/main/java/org/smartregister/maternity/utils/MaternityUtils.java

```diff
@@ -375,4 +381,47 @@ public static void saveMaternityChild(@NonNull List<MaternityEventClient> matern
             Timber.e(e);
         }
     }
+
+    @NonNull
+    public static ContentValues convertMapToContentValues(@Nullable Map<String, String> map) {
+        ContentValues contentValues = new ContentValues();
+
+        if (map != null) {
+            for (Map.Entry<String, String> entry : map.entrySet()) {
+                contentValues.put(entry.getKey(), entry.getValue());
+            }
+        }
+
+        return contentValues;
+    }
+
+    public static void setActionButtonStatus(Button button, String baseEntityId) {
+        button.setTag(R.id.BUTTON_TYPE, R.string.start_maternity);
+        button.setText(R.string.start_maternity);
+
+        String formType = MaternityConstants.EventType.MATERNITY_MEDIC_INFO;
+
+        HashMap<String, String> data = MaternityLibrary.getInstance().getMaternityRegistrationDetailsRepository().findByBaseEntityId(baseEntityId);
+        if (data != null) {
+
+            if ("1".equals(data.get(MaternityConstants.JSON_FORM_KEY.MATERNITY_MEDIC_INFO_SUBMITTED))) {
+                formType = MaternityConstants.EventType.MATERNITY_OUTCOME;
+                button.setText(R.string.outcome);
+                button.setTag(R.id.BUTTON_TYPE, R.string.outcome);
+            }
+        }
+
+        if (MaternityConstants.EventType.MATERNITY_MEDIC_INFO.equals(formType)) {
+            MaternityMedicInfoForm maternityMedicInfoForm = MaternityLibrary.getInstance().getMaternityMedicInfoFormRepository().findOne(new MaternityMedicInfoForm(baseEntityId));
+            if (maternityMedicInfoForm != null) {
+                button.setBackground(ContextCompat.getDrawable(button.getContext(), R.drawable.diagnose_treat_bg));
```

Comment:
you need to create another copy with a descriptive name instead  of `R.drawable.diagnose_treat_bg`

Reply:
file name refactored.